### PR TITLE
Improved deps processing and added to Config

### PR DESCRIPTION
**Pull request by jessemapel**
Updated the deps processing to take json_pointers. I also changed it so you can specify deps at any level. So you can do something like:

```
{
  "instrument" : {
    "pck" : {
      "deps" : ["/base/pck"]
    },
    "deps" : ["/mission"]
  },
  "mission" : {
    ...
  }
}
```

I also inserted the resolution logic into the `Config::evaluateJson` so that it will be applied whenever you try to get an actual set of kernels out.

All of the tests need to be updated so that they go through the Config class instead of calling `getMissionConfig` but I don't know if that belongs in this PR or not.

### CODE REVIEW

**Comment by jessemapel** [comment]
Fixes #79

**Comment by jessemapel** [comment]
Also updated the dependency resolution to allow recursion up to a depth of 10.

File: SpiceQL/tests/KernelTests.cpp [hunk 1]
```diff
@@ -22,13 +22,13 @@ TEST_F(LroKernelSet, UnitTestStackedKernelConstructorDestructor) {
...
     // base LSK still loaded
-    EXPECT_EQ(nkernels, 3);
+    EXPECT_EQ(nkernels, 2);
     EXPECT_EQ(pool.getRefCounts().at(lskPath), 1);
   }
 
   // SCLKs and LSKs are considered text kernels, so they should stay loaded
   ktotal_c("text", &nkernels);
-  EXPECT_EQ(nkernels, 2);
+  EXPECT_EQ(nkernels, 1);
```
Comment (Kelvinrr) [inline SpiceQL/tests/KernelTests.cpp:31]: why did these change?

File: SpiceQL/db/mess.json [hunk 2]
```diff
@@ -36,30 +28,21 @@
...
         "ck" : {
             "reconstructed" : {
-              "kernels" : "msgr_mdis_sc[0-9]{6}_[0-9]{6}_sub_v[0-9]{1}.bc",
-              "deps" : {}
-            },
-            "deps" : {
-                "objs" : ["/base/lsk", "/mess/sclk"]
+              "kernels" : "msgr_mdis_sc[0-9]{6}_[0-9]{6}_sub_v[0-9]{1}.bc"
             }
         }
```
Comment (Kelvinrr) [inline SpiceQL/db/mess.json:40]: Shouldn't these still have `dep` keys pointing to the other frames at the frame level?

**Review by Kelvinrr — COMMENTED** [review]
looks fine, but I think some of the configs are still missing the new style `deps` keys?

File: SpiceQL/tests/KernelTests.cpp (hunk 1 above)
Comment (jessemapel) [reply SpiceQL/tests/KernelTests.cpp:31]: The fixture used to load the SCLKs in its setup and then never unload them. With the change to the time query, I changed it to unload them after setup. So, now there's 1 less text kernel loaded.

File: SpiceQL/db/mess.json (hunk 2 above)
Comment (jessemapel) [reply SpiceQL/db/mess.json:40]: No, the mdis ck deps point over to these. `"deps" : ["/mess/ck", "/mdis_att/ck"]`

Comment (jessemapel) [reply SpiceQL/db/mess.json:40]: I could redo this to better represent the actual frame chain but it'll end up with the same set of kernels in the end and require another step of dependency processing.

**Review by Kelvinrr — APPROVED** [review]

**Pull request merged**